spatialos-sdk/src/worker/component.rs

```diff
@@ -52,17 +52,23 @@ where
 pub(crate) mod internal {
     use crate::worker::internal::schema::*;
     use spatialos_sdk_sys::worker::*;
+    use std::marker::PhantomData;
 
     use crate::worker::component::ComponentId;
 
     #[derive(Debug)]
-    pub struct ComponentData {
+    pub struct ComponentData<'a> {
         pub component_id: ComponentId,
         pub schema_type: SchemaComponentData,
-        pub user_handle: *mut Worker_ComponentDataHandle,
+        pub user_handle: *const Worker_ComponentDataHandle,
+
+        // NOTE: `user_handle` is borrowing data owned by the parent object, but it's a
+        // type-erased pointer that may be null, so we just mark that we're borrowing
+        // *something*.
+        pub _marker: PhantomData<&'a ()>,
```

Comment:
Did not know you could do this. Pretty neat

Reply:
Yup! You actually have to use `PhantomData` if you have a generic parameter that you're not using in the type itself (as is the case here).